### PR TITLE
Migrating Garbage Bin sample from 1.0.0 to 3.1.0

### PR DESCRIPTION
**Pull request by Nirothipan**

### CODE REVIEW

File: garbagebin/component/api/src/main/java/org/homeautomation/garbagebin/api/DeviceTypeService.java [hunk 1]
```diff
@@ -18,69 +18,65 @@
 
 package org.homeautomation.garbagebin.api;
 
-import org.homeautomation.garbagebin.api.dto.DeviceJSON;
-import org.wso2.carbon.apimgt.annotations.api.API;
-import org.wso2.carbon.apimgt.annotations.api.Permission;
-import org.wso2.carbon.device.mgt.extensions.feature.mgt.annotations.DeviceType;
-import org.wso2.carbon.device.mgt.extensions.feature.mgt.annotations.Feature;
+import io.swagger.annotations.*;
```
Comment (charithag) [inline garbagebin/component/api/src/main/java/org/homeautomation/garbagebin/api/DeviceTypeService.java]: Do not use wildcard imports

File: garbagebin/component/api/src/main/java/org/homeautomation/garbagebin/api/DeviceTypeService.java [hunk 2]
```diff
@@ -18,69 +18,65 @@
...
-import javax.ws.rs.Consumes;
-import javax.ws.rs.DELETE;
-import javax.ws.rs.GET;
-import javax.ws.rs.POST;
-import javax.ws.rs.PUT;
-import javax.ws.rs.Path;
-import javax.ws.rs.PathParam;
-import javax.ws.rs.Produces;
-import javax.ws.rs.QueryParam;
+import javax.ws.rs.*;
```
Comment (charithag) [inline garbagebin/component/api/src/main/java/org/homeautomation/garbagebin/api/DeviceTypeService.java]: Do not use wildcard imports

File: SenseMe/feature/feature/src/main/resources/agent/SenseMe.ino [hunk 3]
```diff
@@ -16,100 +16,162 @@
...
 
-#define DHTTYPE DHT11   // DHT 11 
-DHT dht(DHTPIN, DHTTYPE);
+DHT dht(DHT11_PIN, DHTTYPE, 30);
 
+unsigned long previousMillis = 0;        // will store last temp was read
+const long interval = 500;              // interval at which to read sensor
+long startDelay = millis();
 // Update these with values suitable for your network.
+//const char* ssid = "linksys";
```
Comment (charithag) [inline SenseMe/feature/feature/src/main/resources/agent/SenseMe.ino]: Remove un necessary comments

File: SenseMe/feature/feature/src/main/resources/agent/SenseMe.ino [hunk 4]
```diff
@@ -214,55 +273,63 @@ void reconnect() {
...
+            client.subscribe(subscribedTopic);
+            Serial.println("MQTT Client Connected");
+            Serial.print("Subscriped topic : ");
+            Serial.println(subscribedTopic);
+            mqttConnected = true;
+          } else {
+            Serial.println("Error while connecting with MQTT server.");
+          }
+        }
+
```
Comment (charithag) [inline SenseMe/feature/feature/src/main/resources/agent/SenseMe.ino]: Remove unnecessary new lines and format the code

**Review by charithag — CHANGES_REQUESTED** [review]
Please reformat the code by removing unnecessary new lines and comments. Also don't use wildcard imports java in classes.

**Review by charithag — APPROVED** [review]

File: floor-analytics-demo/building-visualizer/agent/BuildingMonitor/BuildingMonito.ino [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+
+#define DHTTYPE DHT11
+
+DHT dht(DHT11_PIN, DHTTYPE, 30); 
+
+unsigned long previousMillis = 0;        // will store last temp was read
+const long interval = 500;              // interval at which to read sensor
+long startDelay = millis();
+// Update these with values suitable for your network.
+//const char* ssid = "linksys";
```
Comment (dakshika) [inline floor-analytics-demo/building-visualizer/agent/BuildingMonitor/BuildingMonito.ino:47]: Remove all commented lines

File: garbagebin/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.garbagebin.operation-bar/operation-bar.hbs [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied. See the License for the
+  specific language governing permissions and limitations
+  under the License.
+}}
+{{#if control_operations}}
+    <div class="wr-operations" style="height: 87px; display: block;">
+        <style>
+            ::-webkit-input-placeholder {
```
Comment (dakshika) [inline garbagebin/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.garbagebin.operation-bar/operation-bar.hbs:21]: Better to add proper class names and override the styles

File: garbagebin/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.garbagebin.operation-bar/operation-bar.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    var autoCompleteParams = context.unit.params.autoCompleteParams;
+    var encodedFeaturePayloads=context.unit.params.encodedFeaturePayloads;
+    var controlOperations = operationModule.getControlOperations(device);
+  //  new Log().info("control operation length : " + controlOperations.length);
+    var queryParams = [];
+    var formParams = [];
+    var pathParams = [];
+    for (var i = 0; i < controlOperations.length; i++) {
+        var currentParamList = controlOperations[i]["params"];
+       // new Log().info("currentParamList length : " + currentParamList.length);
```
Comment (dakshika) [inline garbagebin/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.garbagebin.operation-bar/operation-bar.js:32]: Remove all commented lines

**Review by dakshika — COMMENTED** [review]
need to update according to the feedback

File: floor-analytics-demo/building-visualizer/agent/BuildingMonitor/BuildingMonito.ino (hunk 5 above)
Comment (Nirothipan) [reply floor-analytics-demo/building-visualizer/agent/BuildingMonitor/BuildingMonito.ino:47]: Fixed with https://github.com/wso2/samples-iots/pull/19

File: garbagebin/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.garbagebin.operation-bar/operation-bar.js (hunk 7 above)
Comment (Nirothipan) [reply garbagebin/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.garbagebin.operation-bar/operation-bar.js:32]: Fixed with https://github.com/wso2/samples-iots/pull/19

**Pull request merged**